src/TSHelper.ts

```diff
@@ -200,4 +200,31 @@ export class TSHelper {
         }
         return true;
     }
+
+    // Returns true for expressions that may have effects when evaluated
+    public static isExpressionWithEffect(node: ts.Expression): boolean {
+        return !(ts.isLiteralExpression(node) || ts.isIdentifier(node));
+    }
+
+    // If expression is property/element access with possible effects from being evaluated, returns true along with the
+    // separated object and index expressions.
+    public static isAccessExpressionWithEffects(node: ts.Expression, checker: ts.TypeChecker):
+        [true, ts.Expression, ts.Expression] | [false, null, null] {
```

Comment:
I did this out of habit because I usually use strict null checks and this allows TS to deduce the subsequent types in the tuple by evaluating the first. But you're right, without strict null checks they are equivalent and the later is more succinct.

Reply:
I see. I agree in that case that is actually a nice way of defining the type.